web/src/app/services/draft-survey/draft-survey.service.ts

```diff
@@ -80,30 +85,36 @@ export class DraftSurveyService {
     this.dirty = true;
   }
 
-  addOrUpdateTasks(jobId: string, tasks: List<Task>): void {
+  addOrUpdateTasks(jobId: string, tasks: List<Task>, valid: boolean): void {
     const currentSurvey = this.survey$.getValue();
 
-    const currentJob = currentSurvey.jobs.get(jobId);
+    const currentJob = currentSurvey.jobs.get(jobId)!;
 
-    if (currentJob) {
-      const job = currentJob?.copyWith({
-        tasks: this.dataStoreService.convertTasksListToMap(tasks),
-      });
+    const job = currentJob?.copyWith({
```

Comment:
Do we also need to generate new ids for options, and possibly other nested items? I don't think we depend on options having globally unique ids anywhere, but perhaps we should be consistent since other entities do use unique ids (Surveys, for ex) Perhaps we can file a bug and add a TODO(#issueno) here?

Reply:
I already noticed this. This duplication issue involves options and tasks too. I'll fix it in separate pr.